### ISSUE TITLE
openWith did not work as expected

### ISSUE THREAD

**Issue by fxxjdedd** [microsoft/vscode#106037]
- VSCode Version: v1.48.2
- OS Version: MacOS Catalina 10.15.5

Steps to Reproduce:

1. goto: https://github.com/microsoft/vscode/blob/9f1aa3c9feecd04a79d22fd6752ba14a83b48f1b/src/vs/code/browser/workbench/workbench.ts#L344
2. add:
```
(config.defaultLayout as IDefaultLayout) = {
	editors: [
		{
			uri: URI.file('path/to/README.md').with({ scheme: 'vscode-remote' }),
			openWith: 'vscode.markdown.preview.editor'
		},
	]
};
```
3. get:
![image](https://user-images.githubusercontent.com/13739947/92113057-5542e400-ee21-11ea-949a-d7b8c7087823.png)

it looks like vscode doesn't find the 'vscode.markdown.preview.editor': https://github.com/microsoft/vscode/blob/9f1aa3c9feecd04a79d22fd6752ba14a83b48f1b/src/vs/workbench/contrib/customEditor/browser/customEditors.ts#L256

Does this issue occur when all extensions are disabled?: Yes

**Comment by mjbvz** [comment]
Can you explain what you are trying to do?  Why are you editing the VS Code source?

**Comment by fxxjdedd** [comment]
@mjbvz Thanks for the reply.

I'm using vscode to develop webide, and there is such a requirement: when vscode is started for the first time, a README file is opened by default. By checking the source code and issues[#99283](https://github.com/microsoft/vscode/issues/99283), I found the openWith interface, but the above problem occurred when using it.

**Comment by eamodio** [comment]
Do you have the built-in markdown extension enabled? Since that is what contributes that custom editor.

**Comment by fxxjdedd** [comment]
I believe it is enabled, because I have Markdown options in this way:

![image](https://user-images.githubusercontent.com/13739947/92199906-ff1a8300-eeaa-11ea-8b67-5c13b7f8b157.png)

![image](https://user-images.githubusercontent.com/13739947/92199879-eca04980-eeaa-11ea-9f1d-7666eeef0fb6.png)

I noticed that the placeholder is different, one is `Select editor to use for`, and the other is `Select editor for`, and my method of use is the former, which is to use `CustomEditors` to openWith.

Actually, I debugged a bit in: https://github.com/microsoft/vscode/blob/9f1aa3c9feecd04a79d22fd6752ba14a83b48f1b/src/vs/workbench/contrib/customEditor/browser/customEditors.ts#L256
```
console.log('viewType: ', viewType)
console.log('this._contributedEditors: ', this._contributedEditors)
console.log("this._contributedEditors.get('vscode.markdown.preview.editor'): ", this._contributedEditors.get('vscode.markdown.preview.editor'))
if (!this._contributedEditors.get(viewType)) {
    return this.promptOpenWith(resource, options, group);
}
```
The results are:
![image](https://user-images.githubusercontent.com/13739947/92200545-c7143f80-eeac-11ea-8f1e-a114d7e862e4.png)

The result of synchronously executed code `this._contributedEditors.get` is `undefined`, but when I manually call `temp1.get`in the browser console panel, I can get the `Editor`. So I think it might be a matter of registration timing?
